src/utils/map-controls.js

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// EVENT HANDLING PARAMETERS
+const PITCH_MOUSE_THRESHOLD = 5;
+const PITCH_ACCEL = 1.2;
+const ZOOM_ACCEL = 0.01;
+
+const SUBSCRIBED_EVENTS = [
+  'panstart',
+  'pan',
+  'panend',
+  'pinchstart',
+  'pinch',
+  'pinchend',
+  'doubletap',
+  'wheel'
+];
+
+export default class MapControls {
+  /**
+   * @classdesc
+   * A class that handles events and updates mercator style viewport parameters
+   */
+  constructor() {
+    this.events = SUBSCRIBED_EVENTS;
+  }
+
+  /**
+   * Update the state of the control
+   * @param {MapState} state.mapState - current map state
+   * @param {Function} state.onChangeViewport - callback
+   */
+  setState({mapState, onChangeViewport, isDragging, perspectiveEnabled}) {
+    this.mapState = mapState;
+    this.onChangeViewport = onChangeViewport;
+    this.isDragging = isDragging;
+    this.perspectiveEnabled = perspectiveEnabled;
+  }
+
+  /**
+   * Callback for events
+   * @param {hammer.Event} event
+   */
+  handle(event) {
```

Comment:
Agreed with @ibgreen that this doesn't quite feel right. `InteractiveMap` owns both `EventManager` and `MapControls`, but (with the exception of instantiation and destruction) it doesn't really do anything with `EventManager`.

I think the intent here is for both `EventManager` and `MapControls` to be exposed for custom implementation. However:
- `EventManager` is not specified as a prop on `InteractiveMap` so it can't currently be overridden / extended;
- `InteractiveMap` probably doesn't need to care about `EventManager` at all, with the exception of exposing it to application developers. So it could be passed through to `MapControls`; we'd then also need a `destroy()` hook on `MapControls` to destroy the `EventManager` as well.

I'd like to see this straightened out before landing...think it's almost there.